panel/param.py

```diff
@@ -456,6 +456,10 @@ def widget(self, p_name):
                         widget_class = LiteralInput
             if hasattr(widget_class, 'step') and getattr(p_obj, 'step', None):
                 kw['step'] = p_obj.step
+            if hasattr(widget_class, 'fixed_start') and getattr(p_obj, 'bounds', None):
```

Comment:
I'll have to consider this logic more deeply, I don't think this is actually sufficient since it's the `softbounds` that we want to apply.

Reply:
I'm not sure I understand what you are meaning. The `softbounds` functionality for the slider is already implemented, and the `hard/fixed/..._*` should be associated with the hard `bounds`, right?

https://user-images.githubusercontent.com/19758978/183444918-4cbb8465-6954-49ca-a787-22d4124673bc.mp4

``` python
import panel as pn
import param
import pytest

pn.extension()

class Test(param.Parameterized):
    i = param.Number(default=1, softbounds=(1, 5), bounds=(0, 10), step=1)


t = Test()
w = pn.widgets.EditableFloatSlider.from_param(t.param.i)
w
```

--- 
Unrelated to this PR, I noted that `bounds` / `softbounds` have priority over `start`/`end`. Is this what is expected? 

![image](https://user-images.githubusercontent.com/19758978/183446935-ec21f191-a630-47e2-a5e7-31a8c860c8fd.png)

Reply:
Yes, putting in a value outside the soft bounds but inside the hard bounds is precisely how you can make use of the difference between soft and hard bounds.

Reply:
That does mean that it's not possible to reset the soft bounds to their defaults.

Reply:
Yes. But still better than being bound inappropriately by soft bounds, as one can usually reload the app, but can never hack around falsely hard bounds.

Reply:
See https://github.com/bokeh/bokeh/issues/9943 for my rant about this.

Reply:
Oh, I see you chimed in there before even joining our group. :-)